### DISCUSSION TITLE
How to nest markers inside a parent marker?

### DISCUSSION THREAD

**Discussion by FeryET** [pytest-dev/pytest#11331] [Q&A] [answered]
Hi.

I'm trying to come up with a solution to have more finegrained marking for my tests.

```python
import pytest

@pytest.mark.sub.pass
def test_pass():
    pass

@pytest.mark.sub.fail
def test_fail():
    raise Exception()

```

I want to have a parent maker like `sub` above that has multiple submarkers. Is this possible using pytest?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
Currently markers don't have that feature

Without a compelling concrete use cases I'm not convinced of that feature

**Comment by The-Compiler** [comment]
Note you could DIY a thing like this as well:

```python
import pytest

class MarkerGen:

    def __init__(self, parts=None):
        self.parts = parts or []

    def __getattr__(self, name):
        return MarkerGen(self.parts + [name])

    def __call__(self, *args, **kwargs):
        path = ".".join(self.parts)
        marker = getattr(pytest.mark, path)
        return marker(*args, **kwargs)


mark = MarkerGen()

@mark.sub.passing
def test_pass():
    pass

@mark.sub.failing
def test_fail():
    raise Exception()
```

Using `.` as a separator might be a bit unorthodox, but works thanks to `getattr()`, so you can select the two tests with `-m sub.passing` and `-m sub.failing` respectively.
